### PR TITLE
refactor(lib): migrate remaining "Request" structs to "Methods" that use an Instance

### LINKED ISSUE #1235: cleanup: Rename lib methods containers away from "Requests"
The objects that contain the interface of lib used to be "[something]Requests" like DatasetRequests. We renamed some, such as ProfileMethods (used to be ProfileRequests), but did not finish the job.

Take a look at the function `Receivers` in `lib/lib.go`. All of the objects constructed in there should be using a name that ends in "Methods". We should also rename the `CoreRequestsName` interface function to `CoreMethodsName`.

### PR DESCRIPTION
**Pull request by b5**
closes #1235

### CODE REVIEW

File: cmd/save_test.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func TestSaveRun(t *testing.T) {
 
 	for _, c := range cases {
 		run.IOReset()
-		dsr, err := f.DatasetRequests()
+		dsr, err := f.DatasetMethods()
```
Comment (dustmop) [inline cmd/save_test.go:143]: dsr -> dsm

File: cmd/rename_test.go [hunk 2]
```diff
@@ -122,17 +122,17 @@ func TestRenameRun(t *testing.T) {
 	}
 
 	for i, c := range cases {
-		dsr, err := f.DatasetRequests()
+		dsr, err := f.DatasetMethods()
```
Comment (dustmop) [inline cmd/rename_test.go:125]: dsr -> dsm

File: cmd/dag_test.go [hunk 3]
```diff
@@ -145,15 +145,15 @@ func TestDAGInfo(t *testing.T) {
 		//     },
 	}
 	for i, c := range cases {
-		dsr, err := f.DatasetRequests()
+		dsr, err := f.DatasetMethods()
```
Comment (dustmop) [inline cmd/dag_test.go:148]: dsr -> dsm

File: cmd/diff_test.go [hunk 4]
```diff
@@ -87,14 +87,14 @@ func TestDiffRun(t *testing.T) {
 
 	for _, c := range good {
 		t.Run(c.description, func(t *testing.T) {
-			dsr, err := f.DatasetRequests()
+			dsr, err := f.DatasetMethods()
```
Comment (dustmop) [inline cmd/diff_test.go:90]: dsr -> dsm

File: cmd/diff_test.go [hunk 5]
```diff
@@ -119,7 +119,7 @@ func TestDiffRun(t *testing.T) {
 	}
 
 	for _, c := range bad {
-		dsr, err := f.DatasetRequests()
+		dsr, err := f.DatasetMethods()
```
Comment (dustmop) [inline cmd/diff_test.go:122]: dsr -> dsm

File: cmd/remove_test.go [hunk 6]
```diff
@@ -106,17 +106,17 @@ func TestRemoveRun(t *testing.T) {
 	}
 
 	for i, c := range cases {
-		dsr, err := f.DatasetRequests()
+		dsr, err := f.DatasetMethods()
```
Comment (dustmop) [inline cmd/remove_test.go:109]: dsr -> dsm

File: cmd/stats_test.go [hunk 7]
```diff
@@ -82,7 +82,7 @@ func TestStatsRun(t *testing.T) {
 		t.Fatalf("error creating new test factory: %s", err)
 	}
 
-	dsr, err := f.DatasetRequests()
+	dsr, err := f.DatasetMethods()
```
Comment (dustmop) [inline cmd/stats_test.go:85]: dsr -> dsm

File: cmd/validate_test.go [hunk 8]
```diff
@@ -91,19 +91,19 @@ func TestValidateRun(t *testing.T) {
 	}
 
 	for i, c := range cases {
-		dsr, err := f.DatasetRequests()
+		dsr, err := f.DatasetMethods()
```
Comment (dustmop) [inline cmd/validate_test.go:94]: dsr -> dsm

File: lib/datasets.go [hunk 9]
```diff
@@ -32,44 +32,29 @@ import (
...
+type DatasetMethods struct {
 	// TODO (b5) - remove cli & node fields in favour of inst accessors:
 	cli  *rpc.Client
 	node *p2p.QriNode
 	inst *Instance
 }
 
 // CoreRequestsName implements the Requets interface
-func (DatasetRequests) CoreRequestsName() string { return "datasets" }
+func (DatasetMethods) CoreRequestsName() string { return "datasets" }
```
Comment (dustmop) [inline lib/datasets.go:44]: Could also like if we could change this method name to CoreMethodsName, though that also touches a lot of files

Comment (b5) [reply lib/datasets.go:44]: totally agreed, I think it's worth a follow-up PR

**Pull request merged**